sylvia/tests/reply_dispatch.rs

```diff
@@ -0,0 +1,371 @@
+        self.last_reply
+            .save(ctx.deps.storage, &REMOTE_INSTANTIATED_REPLY_ID)?;
+        let init_data = parse_instantiate_response_data(&data.unwrap())?;
+        let remote_addr = Addr::unchecked(init_data.contract_address);
+
+        self.remote
+            .save(ctx.deps.storage, &Remote::new(remote_addr))?;
+
+        Ok(Response::new())
+    }
+
+    #[sv::msg(reply, handlers=[success, both], reply_on=success)]
+    fn success(
+        &self,
+        ctx: ReplyCtx<Q>,
+        #[sv::data(raw, opt)] _data: Option<Binary>,
+        #[sv::payload(raw)] _payload: Binary,
+    ) -> Result<Response<M>, ContractError> {
+        self.last_reply.save(ctx.deps.storage, &SUCCESS_REPLY_ID)?;
+
+        Ok(Response::new())
+    }
+
+    #[sv::msg(reply, handlers=[error, both], reply_on=error)]
+    fn error(
+        &self,
+        ctx: ReplyCtx<Q>,
+        _error: String,
+        #[sv::payload(raw)] _payload: Binary,
+    ) -> Result<Response<M>, ContractError> {
+        self.last_reply.save(ctx.deps.storage, &ERROR_REPLY_ID)?;
+
+        Ok(Response::new())
+    }
+
+    #[sv::msg(reply, reply_on=always)]
+    fn always(
+        &self,
+        ctx: ReplyCtx<Q>,
+        _result: SubMsgResult,
+        #[sv::payload(raw)] _payload: Binary,
+        // _first_part_payload: u32,
+        // _second_part_payload: String,
+    ) -> Result<Response<M>, ContractError> {
+        self.last_reply.save(ctx.deps.storage, &ALWAYS_REPLY_ID)?;
+
+        Ok(Response::new())
+    }
+
+    #[sv::msg(exec)]
+    fn send_cosmos_messages(&self, ctx: ExecCtx<Q>) -> Result<Response<M>, ContractError> {
+        let remote_addr = self.remote.load(ctx.deps.storage)?;
+        let cosmos_msg = CosmosMsg::Bank(BankMsg::Send {
+            to_address: remote_addr.as_ref().to_string(),
+            amount: vec![],
+        });
+        let submsg = cosmos_msg.always(Binary::default())?;
+        Ok(Response::new().add_submessage(submsg))
+    }
+}
+
+mod tests {
+    use crate::noop_contract::sv::mt::CodeId as NoopCodeId;
+    use crate::sv::mt::{CodeId, ContractProxy};
+    use crate::sv::{
+        ALWAYS_REPLY_ID, ERROR_REPLY_ID, REMOTE_INSTANTIATED_REPLY_ID, SUCCESS_REPLY_ID,
+    };
+
+    use sylvia::cw_multi_test::IntoBech32;
+    use sylvia::cw_std::StdError;
+    use sylvia::multitest::App;
+
+    #[test]
+    fn dispatch_replies() {
+        let app = App::default();
+        let code_id = CodeId::store_code(&app);
+        let noop_code_id = NoopCodeId::store_code(&app);
+
+        let owner = "owner".into_bech32();
+
+        // Trigger remote instantiation reply
+        let contract = code_id
+            .instantiate(noop_code_id.code_id())
+            .with_label("Contract")
+            .call(&owner)
+            .unwrap();
+        let last_reply = contract.last_reply().unwrap();
+        assert_eq!(last_reply, REMOTE_INSTANTIATED_REPLY_ID);
+
+        // Should not dispatch if expected success and execution failed
```

Comment:
All those comments should be turned into separate test cases

Reply:
This checks are so small that I prefer to keep them in the same test case. There is no benefit in the readability in splitting them.